nautobot_circuit_maintenance/jobs/site_search.py

```diff
@@ -144,38 +124,45 @@ def run(self, data=None, commit=None):
             # Check to see if there are any circuit maintenances that are overlapping at some moment in time
-            for site in site_list:
+            for site in site_set:
                 for other_circuit_maint in circuit_maintenance_mapper[site.name]:
                     # Report failures for any time where a circuit will take an outage
-                    if circuit_maint == other_circuit_maint:
+                    # Check to see if the other circuit maintenance has already had overlap checked.
+                    # If this is the same circuit maintenance record, that will also show up in the processed list
+                    # since the list is being added to before the loop.
+                    if other_circuit_maint in processed_maintenance_list:
                         continue
                     if check_for_overlap(circuit_maint, other_circuit_maint):
+                        overlapping_maintenance = True
                         if PLUGIN_SETTINGS.get("overlap_job_exclude_no_impact"):
                             for impact in circuit_maint.circuitimpact_set:
                                 if impact.impact != "NO-IMPACT":
                                     self.log_warning(
                                         obj=site,
-                                        message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
+                                        message=f"There is an overlapping __no-impact__ maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
                                     )
                         else:
                             self.log_warning(
                                 obj=site,
                                 message=f"There is an overlapping maintenance for site: {site.name} on {circuit_maint.start_time}. Other maintenances: {other_circuit_maint}|{circuit_maint}",
                             )
-                    else:
-                        # Log success for each time there is a known circuit still available at the site at the same time
-                        self.log_debug(message="Checked maintenance for overlap, no overlap was found.")
+
+            # Log success for each time there is a known circuit still available at the site at the same time
+            if overlapping_maintenance is False:
+                if data["job_debug"]:
+                    self.log_info(obj=circuit_maint, message="Checked maintenance for overlap, no overlap was found.")
 
         self.log_success(
             obj=None,
-            message=f"Successfully checked through {counter} maintenance notification{'s' if counter > 1 else ''}.",
+            message=f"Successfully checked through {circuit_maintenances.count()} maintenance notification{'s'[:circuit_maintenances.count()^1]}.",
```

Comment:
sorry, but I don't get `^1`, I tried it and I don't understand the logic

Reply:
Bit wise "magic". @gsnider2195

Reply:
Basically if the count is 1 the result is 0 which will return nothing in the string. If there is any other integer count, then it returns the rest of the string.

Reply:
It's XORing the count. 1 returns 0, everything else returns 1. Then it's using that to slice the string. So:

```
checked through 0 maintenance notifications.
checked through 1 maintenance notification.
checked through 2 maintenance notifications.
```

Reply:
> Basically if the count is 1 the result is 0 which will return nothing in the string. If there is any other integer count, then it returns the rest of the string.

I didn't see this reply. This is the answer 😄